### PR TITLE
Adding brief parameter to humanize

### LINKED ISSUE #1001: Brief parameter for humanize function
## Feature Request

This parameter is False by default. When true, it shortens the date and time terms. For example:
```
>>> present = arrow.utcnow()
>>> shift = present.shift(hours=2, minutes=34, seconds=56)
>>> shift.humanize(present, brief=True, only_distance=True)
>>> 2h 34m 56s
```

### PR DESCRIPTION
**Pull request by akotha7**
## Pull Request Checklist

Thank you for taking the time to improve Arrow! Before submitting your pull request, please check all *appropriate* boxes:

- [x] 🧪  Added **tests** for changed code.
- [x] 🛠️  All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [x] 🧹  All linting checks **pass** when run locally (run `tox -e lint` or `make lint` to find out!).
- [x] 📚  Updated **documentation** for changed code.
- [x] ⏩  Code is **up-to-date** with the `master` branch.

If you have *any* questions about your code changes or any of the points above, please submit your questions along with the pull request and we will try our best to help!

## Description of Changes

- Added `brief` optional parameter to `humanize` function, based off #1001 
- Added associated tests inside `test_arrow.py`
- Updated documentation to reflect new feature

## Questions
Current implementation shortens into `only_distance` and then each term. One edge case that occurs for example is "an hour and 6 minutes" should that become "h 6m" or "1h 6m"

Closes #1001

### CODE REVIEW

**Comment by anishnya** [comment]
@jadchaar @krisfremen @systemcatch, what are your thoughts here?

**Comment by systemcatch** [comment]
@akotha7 I tend to think we should use the `1h 6m` as this seems more consistent to me.

i will try to look this over properly in the next few days.

File: docs/index.rst [hunk 1]
```diff
@@ -243,6 +242,20 @@ Indicate a specific time granularity (or multiple):
     >>> future.humanize(present, only_distance=True, granularity=["hour", "minute"])
     'an hour and 6 minutes'
 
+Indicate shortened time:
```
Comment (systemcatch) [inline docs/index.rst:245]: How about `Humanize in a shortened time format`?

File: docs/index.rst [hunk 2]
```diff
@@ -243,6 +242,20 @@ Indicate a specific time granularity (or multiple):
...
+
+    >>> present = arrow.utcnow()
+    >>> future = present.shift(hours=2)
+    >>> future.humanize(present)
+    'in 2 hours'
+    >>> future.humanize(present, brief=True)
+    '2h'
+    >>> future = present.shift(minutes=66)
+    >>> present.humanize(future, granularity=["hour", "minute"], brief=True)
+    'h 6m'
```
Comment (systemcatch) [inline docs/index.rst:257]: Can't say I'm a fan of dropping the `1` here.

File: arrow/locales.py [hunk 3]
```diff
@@ -285,6 +295,18 @@ def _format_relative(
...
+        humanized: str,
+    ) -> str:
+
+        parts = humanized.split()
+
+        if parts[0].isnumeric():
+            return parts[0] + parts[1][0]
+
+        return parts[1][0]
+
```
Comment (systemcatch) [inline arrow/locales.py:309]: Could there be a case for creating a `timeframes_brief` dict for each locale that supports this feature?

I'm not sure how locales outside of English deal with shortening.

**Review by systemcatch — CHANGES_REQUESTED** [review]
First pass.

It would be nice to have a test where using the brief parameter fails in some way.

File: docs/index.rst (hunk 2 above)
Comment (LesterKim) [reply docs/index.rst:257]: ^ Ditto

File: arrow/locales.py [hunk 4]
```diff
@@ -285,6 +295,18 @@ def _format_relative(
 
         return direction.format(humanized)
 
+    def _format_brief(
+        self,
+        humanized: str,
+    ) -> str:
+
```
Comment (LesterKim) [inline arrow/locales.py:302]:
```suggestion
    def _format_brief(self, humanized: str) -> str:
```

File: arrow/locales.py [hunk 5]
```diff
@@ -138,41 +138,51 @@ def describe(
...
-        if self.and_word:
+        if self.and_word and not brief:
             parts.insert(-1, self.and_word)
         humanized = " ".join(parts)
 
         if not only_distance:
             humanized = self._format_relative(humanized, *timeframes[-1])
+        if brief:
+            parts = [self._format_brief(part) for part in parts]
+            humanized = " ".join(parts)
```
Comment (LesterKim) [inline arrow/locales.py:185]:
```suggestion
            return " ".join(self._format_brief(part) for part in parts)
```

File: arrow/locales.py [hunk 6]
```diff
@@ -138,41 +138,51 @@ def describe(
...
         """
 
         humanized = self._format_timeframe(timeframe, trunc(delta))
         if not only_distance:
             humanized = self._format_relative(humanized, timeframe, delta)
+        if brief:
+            humanized = self._format_timeframe(timeframe, trunc(delta))
+            humanized = self._format_brief(humanized)
 
         return humanized
```
Comment (LesterKim) [inline arrow/locales.py:158]:
```suggestion
        humanized = self._format_timeframe(timeframe, trunc(delta))

        if not only_distance:
            humanized = self._format_relative(humanized, timeframe, delta)
        if brief:
            humanized = self._format_timeframe(timeframe, trunc(delta))
            humanized = self._format_brief(humanized)

        return humanized
```

File: arrow/locales.py [hunk 7]
```diff
@@ -138,41 +138,51 @@ def describe(
...
             parts.insert(-1, self.and_word)
         humanized = " ".join(parts)
 
         if not only_distance:
             humanized = self._format_relative(humanized, *timeframes[-1])
+        if brief:
+            parts = [self._format_brief(part) for part in parts]
+            humanized = " ".join(parts)
 
         return humanized
```
Comment (LesterKim) [inline arrow/locales.py:187]:
```suggestion

        humanized = " ".join(parts)

        if brief:
            return " ".join(self._format_brief(part) for part in parts)

        if only_distance:
            return humanized

        return self._format_relative(humanized, *timeframes[-1])
```

**Review by LesterKim — CHANGES_REQUESTED** [review]
I added suggestions to make the logic easier to follow.
